lib/services/access_request.go

```diff
@@ -60,6 +60,44 @@ func ValidateAccessRequest(ar types.AccessRequest) error {
 	return nil
 }
 
+// ClusterGetter provides access to the local cluster
+type ClusterGetter interface {
+	// GetClusterName returns the local cluster name
+	GetClusterName(opts ...MarshalOption) (types.ClusterName, error)
+	// GetRemoteCluster returns a remote cluster by name
+	GetRemoteCluster(clusterName string) (types.RemoteCluster, error)
+}
+
+// ValidateAccessRequestClusterNames checks that the clusters in the access request exist
+func ValidateAccessRequestClusterNames(cg ClusterGetter, ar types.AccessRequest) error {
+	localClusterName, err := cg.GetClusterName()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	var errs []error
+	for _, resourceID := range ar.GetRequestedResourceIDs() {
+		if resourceID.ClusterName == "" {
+			continue
+		}
+		if resourceID.ClusterName == localClusterName.GetClusterName() {
+			continue
+		}
+		_, err := cg.GetRemoteCluster(resourceID.ClusterName)
+		if err == nil {
+			continue
+		}
+		if trace.IsNotFound(err) {
```

Comment:
When I asked about handling "not found" before, I didn't mean to ignore everything else but rather return.

```go
_, err = cg.GetRemoteCluster(resourceID.ClusterName)
if err != nil && !trace.IsNotFound(err) {
  return trace.Wrap(err, "failed to fetch remote cluster %q", resourceID.ClusterName)
}
if trace.IsNotFound(err) {
  // add cluster name to the list of invalid clusters
}
```

Reply:
Ah that makes more sense than how i first read it.